Makefile.am

```diff
@@ -215,6 +215,15 @@ _usnic_files = \
 
 if HAVE_VERBS
 _usnic_files += prov/usnic/src/usdf_fake_ibv.c
+
+# If the verbs or usnic providers are being built as a DL, then we
+# need to link libibverbs into the usnic provider
+if HAVE_VERBS_DL
+_usnic_libibverbs = -libverbs
+endif
+if HAVE_USNIC_DL
+_usnic_libibverbs = -libverbs
+endif
 endif
```

Comment:
@pmmccorm We actually need it for both cases -- the HAVE_VERBS_DL case is the case you identified (e.g., `--enable-usnic=yes --enable-verbs=dl`).  The HAVE_USNIC_DL case is for `--enable-usnic=dl` case.

@goodell Sure, I can move the "or" into configure.m4 logic (sadly, there's no "if HAVE_VERBS_DL || HAVE_USNIC_DL" Automake syntax) and use a single new AM_CONDITIONAL for it.